### ISSUE TITLE
FMX

### ISSUE THREAD

**Issue by jkour** [WladiD/AnyiQuack#6]
Hi,
that's great library.
Would it work in Firemonkey? Or can it be converted?
Thanks

**Comment by WladiD** [comment]
Hi John,

thanks for your kudos.

AnyiQuack works already with FMX, see the included example SlidingButtonFMX.

Best regards from Germany!

**Comment by jkour** [comment]
Hi,
I missed that demo as it is not part of the examples project group :-)
You use Winapi units so they do not compile for other platforms sadly

**Comment by WladiD** [comment]
Hi,

you are right, currently it works only under Windows. Because this is my main target platform. But I'm looking forward in this aspect.

**Comment by jkour** [comment]
Hi,
thanks for the reply.
Would you check a pull request for FMX?

**Comment by WladiD** [comment]
Yes, I would like to do that.

**Comment by WladiD** [comment]
Hi John,

thank you very much for this big improvement of AnyiQuack.

**Comment by jkour** [comment]
@WladiD My pleasure :-)

I was testing the FMX code on mac and Android and it works really smoothly. That is great work.

Quick Question: You choose Synchronize over Queue. Wouldn't Queue be more appropriate given that it is not blocking the app? I would expect this for the case of multiple animations running in parallel

**Comment by WladiD** [comment]
The Synchronize is blocking the TTimerThread, so it is not firing next time events to the main thread until the current one is processed.

For example: If your main thread is busy for 150ms, with the Queue solution you get 10 entries in the main thread to execute (15ms resolution) which are then executed when the main thread is getting idle. With the Synchronize solution it waits 150ms and is then called once (for the span of the 150ms example).

I've got perfect experience in the past with SendMessageTimeout instead of SendMessage. This is compareable with Synchronize/Queue.

**Comment by jkour** [comment]
Yes; I see the difference.

Thanks
